core/src/impl/Kokkos_Profiling_Interface.cpp

```diff
@@ -52,6 +52,20 @@ namespace Kokkos {
        	return (NULL != initProfileLibrary);
     }
 
+    void pushRegion(const std::string& kName) {
+	if( NULL != pushRegionCallee ) {
+	    Kokkos::fence();
```

Comment:
What is the usage-context of the profiling regions that would be at risk for a memory race?
Will different threads be setting and using the push & pop?
Would be best to have a short comment.